extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/LambdaUtil.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.amazon.lambda.deployment;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import io.quarkus.deployment.pkg.builditem.OutputTargetBuildItem;
+
+public class LambdaUtil {
+
+    /**
+     * Strips period, dash, and numbers. Turns characters after to uppercase. i.e.
+     * Also strips "-SNAPSHOT" from end of name.
+     *
+     * "foo.bar-1.0-SNAPSHOT" to "FooBar"
+     *
+     * @param basename
+     * @return
+     */
+    public static String artifactToLambda(String basename) {
+        if (basename.endsWith("-SNAPSHOT"))
+            basename = basename.substring(0, basename.length() - "-SNAPSHOT".length());
+        String name = convertToken(basename, "[.0-9-]");
+        return name.trim();
+    }
+
+    protected static String convertToken(String basename, String token) {
+        String[] splits = basename.split(token);
+        if (splits == null || splits.length == 0)
+            return basename;
+        String name = "";
+        for (String split : splits) {
+            split = split.trim();
+            if (split.isEmpty())
+                continue;
+            name = name + split.substring(0, 1).toUpperCase() + split.substring(1).toLowerCase();
+        }
+        return name;
+    }
+
+    public static void writeFile(OutputTargetBuildItem target, String name, String output) throws IOException {
+        FileOutputStream os = new FileOutputStream(target.getOutputDirectory().resolve(name).toFile());
```

Comment:
This should be in a try with resources block, or better yet just use Files.write()